lib/cyrusdb_twoskip.c

```diff
@@ -976,9 +973,7 @@ static int relocate(struct dbengine *db)
             loc->forwardloc[i] = _getloc(db, &loc->record, i);
 
         /* make sure this record is complete */
-        r = check_tailcrc(db, &loc->record);
-
-        if (r) return r;
+        return check_tailcrc(db, &loc->record);
     }
```

Comment:
Same complaint I had on the skiplist one... I don't like this style of cleanup generally, because it makes the final if() block behave specially with regards to returning early on non-failure cases, and this handling of 'r' will need to be re-added later if another block gets placed afterwards.